### PR TITLE
PR: Test using `PyQt` extra packages

### LINKED ISSUE #383: Add extra packages available for PyQt to CI for testing
Follow-up issue as discussed at https://github.com/spyder-ide/qtpy/pull/372 to add to the test suite setup extra packages

### PR DESCRIPTION
**Pull request by dalthviz**
* Add CI matrix value to install PyQt extra packages
* Remove setting specific QScintilla version for PyQt6 and instead just install it without version constraint when using the pyqt-extras flag: checking the CI logs the test was being skipped for errors when trying to import `QSci` due to DLL errors or because the specified version was not compatible with the PyQt6 version being installed and then the import failed.
* Add step to review jobs matrix/config values
* Move some jobs to run using latest Qt6 bindings (6.5)
* Update README with testing matrix value section and some improvements to the typing section and the general content

Fixes #383

### CODE REVIEW

File: .github/workflows/test.sh
Comment (ccordoba12) [inline .github/workflows/test.sh:39]:
```suggestion
            pip install pyqt5==${PYQT5_VERSION}.* \
            PyQtWebEngine==${PYQT5_VERSION}.* \
            QScintilla==${QSCINTILLA_VERSION}.* \
            PyQt3D==${PYQT5_VERSION}.* \
            PyQtChart==${PYQT5_VERSION}.* \
            PyQtDataVisualization==${PYQT5_VERSION}.* \
            PyQtNetworkAuth==${PYQT5_VERSION}.* \
            PyQtPurchasing==${PYQT5_VERSION}.*
```

To make this easier to read.

File: .github/workflows/test.sh
Comment (ccordoba12) [inline .github/workflows/test.sh:41]:
```suggestion
            pip install pyqt5==${PYQT5_VERSION}.* \
            PyQtWebEngine==${PYQT5_VERSION}.* \
            QScintilla==${QSCINTILLA_VERSION}.*
```

File: .github/workflows/test.sh
Comment (ccordoba12) [inline .github/workflows/test.sh:47]:
```suggestion
            pip install pyqt6==${PYQT6_VERSION}.* \
            PyQt6-WebEngine==${PYQT6_VERSION}.* \
            PyQt6-Qt6==${PYQT6_QT_VERSION}.* \
            PyQt6-QScintilla \
            PyQt6-3D==${PYQT6_VERSION}.* \
            PyQt6-Charts==${PYQT6_VERSION}.* \
            PyQt6-DataVisualization==${PYQT6_VERSION}.* \
            PyQt6-NetworkAuth==${PYQT6_VERSION}.*
```

File: .github/workflows/test.sh
Comment (ccordoba12) [inline .github/workflows/test.sh:49]:
```suggestion
            pip install pyqt6==${PYQT6_VERSION}.* \
            PyQt6-WebEngine==${PYQT6_VERSION}.* \
            PyQt6-Qt6==${PYQT6_QT_VERSION}.*
```

File: README.md
Comment (ccordoba12) [inline README.md:157]:
```suggestion
**Note**: These steps are necessary for the correct usage of the default VSCode's type
checking feature while using QtPy in your source code.
```

File: README.md
Comment (ccordoba12) [inline README.md:162]:
```suggestion
Currently, QtPy runs tests for different bindings on Linux, Windows and macOS, using
Python 3.7 and 3.11, and installing those bindings with `conda` and `pip`. For the
PyQt bindings, we also check the installation of extra packages via `pip`.
```

File: README.md
Comment (ccordoba12) [inline README.md:168]:
```suggestion
| OS      | Binding / manager | conda                                      | pip  | conda              | pip                        |
```

File: README.md
Comment (ccordoba12) [inline README.md:182]:
```suggestion
The mentioned extra packages for the PyQt bindings are the following:
```

File: README.md
Comment (ccordoba12) [inline README.md:193]:
```suggestion
Everyone is welcome to contribute! See our [Contributing guide](CONTRIBUTING.md) for more details.
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks @dalthviz! Some small suggestions to improve readability and wording, the rest looks good to me.

**Comment by dalthviz** [comment]
Thanks for the review @ccordoba12 ! As side note, probably it could be better to wait for https://github.com/spyder-ide/qtpy/pull/451 to get merged before merging this, right?

File: README.md
Comment (ccordoba12) [inline README.md:146]:
```suggestion
in `pyproject.toml`. See [here](https://github.com/microsoft/pyright/blob/main/docs/configuration.md)
for reference. In order to set this configuration, QtPy offers the `pyright-config`
command for guidance.
```

File: README.md
Comment (ccordoba12) [inline README.md:157]:
```suggestion
have them, it is recommended to create the latter. For example, in an environment where PyQt5
is installed and selected (or the default fallback, if no binding can be found in the
environment), this would output the following:
```

**Review by ccordoba12 — COMMENTED** [review]
Last suggestions, then this should ready for merge.

**Comment by ccordoba12** [comment]
> As a side note, probably it could be better to wait for https://github.com/spyder-ide/qtpy/pull/451 to get merged before merging this, right?

#451 is going to touch a lot of things, so it could be difficult to merge this after that one.

@Czaki, what do you think?

**Comment by Czaki** [comment]
At this moment, I do not see conflicts between these two packages. Is some happen (formating) then execute black should easily solve them.

**Review by ccordoba12 — APPROVED** [review]
Thanks @dalthviz!

**Pull request merged**
